File: .sling/libs/web5/eqela/jk-net/0.38.0/jk.dns.js
let JkDnsDNSResolver = function() {};

JkDnsDNSResolver.prototype._t = { "JkDnsDNSResolver" : true };
JkDnsDNSResolver.prototype._tobj = JkDnsDNSResolver;

JkDnsDNSResolver.NEW = function() {
	var v = new JkDnsDNSResolver;
	return v.CTOR_JkDnsDNSResolver();
};

JkDnsDNSResolver.prototype.CTOR_JkDnsDNSResolver = function() {
	return this;
};

JkDnsDNSResolver.create = function() {
	return null;
};

JkDnsDNSResolver.instance = function() {
	return JkDnsDNSResolver.create();
};

JkDnsDNSResolver.prototype.getNSRecords = function(host, type, ctx) {};

JkDnsDNSResolver.prototype.getIPAddress = function(hostname, ctx) {};

JkDnsDNSResolver.prototype.getIPAddresses = function(hostname, ctx) {};

JkDnsDNSResolver.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkDnsDNSResolver"] === true;
};

let JkDnsDNSRecord = function() {
	this.ttl = 0;
	this.address = null;
};

JkDnsDNSRecord.prototype._t = {
	"JkLangStringObject" : true,
	"JkDnsDNSRecord" : true
};
JkDnsDNSRecord.prototype._tobj = JkDnsDNSRecord;

JkDnsDNSRecord.NEW = function() {
	var v = new JkDnsDNSRecord;
	return v.CTOR_JkDnsDNSRecord();
};

JkDnsDNSRecord.prototype.CTOR_JkDnsDNSRecord = function() {
	this.address = null;
	this.ttl = 0;
	return this;
};

JkDnsDNSRecord.prototype.getType = function() {
	return "UNKNOWN";
};

JkDnsDNSRecord.prototype.toString = function() {
	return "type=" + (JkLangString.safeString((this.getType()))) + (" ttl=") + (JkLangString.safeString((JkLangString.forInteger(this.ttl)))) + (" address=`") + (JkLangString.safeString(this.address)) + ("'");
};

JkDnsDNSRecord.prototype.getTtl = function() {
	return this.ttl;
};

JkDnsDNSRecord.prototype.setTtl = function(v) {
	this.ttl = v;
	return this;
};

JkDnsDNSRecord.prototype.getAddress = function() {
	return this.address;
};

JkDnsDNSRecord.prototype.setAddress = function(v) {
	this.address = v;
	return this;
};

JkDnsDNSRecord.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkDnsDNSRecord"] === true;
};

let JkDnsDNSRecordNS = function() {
	JkDnsDNSRecord.call(this);
};

JkDnsDNSRecordNS.prototype = (function(o) { function F() {}; F.prototype = o; return(new F()); }) (JkDnsDNSRecord.prototype);
JkDnsDNSRecordNS.prototype.constructor = JkDnsDNSRecordNS;
JkDnsDNSRecordNS.prototype._t = {
	"JkDnsDNSRecordNS" : true,
	"JkLangStringObject" : true,
	"JkDnsDNSRecord" : true
};
JkDnsDNSRecordNS.prototype._tobj = JkDnsDNSRecordNS;

JkDnsDNSRecordNS.NEW = function() {
	var v = new JkDnsDNSRecordNS;
	return v.CTOR_JkDnsDNSRecordNS();
};

JkDnsDNSRecordNS.prototype.CTOR_JkDnsDNSRecordNS = function() {
	if(JkDnsDNSRecord.prototype.CTOR_JkDnsDNSRecord.call(this) == null) {
		return null;
	}
	return this;
};

JkDnsDNSRecordNS.prototype.getType = function() {
	return "NS";
};

JkDnsDNSRecordNS.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkDnsDNSRecordNS"] === true;
};

let JkDnsDNSRecordMX = function() {
	JkDnsDNSRecord.call(this);
	this.priority = 0;
};

JkDnsDNSRecordMX.prototype = (function(o) { function F() {}; F.prototype = o; return(new F()); }) (JkDnsDNSRecord.prototype);
JkDnsDNSRecordMX.prototype.constructor = JkDnsDNSRecordMX;
JkDnsDNSRecordMX.prototype._t = {
	"JkDnsDNSRecordMX" : true,
	"JkLangStringObject" : true,
	"JkDnsDNSRecord" : true
};
JkDnsDNSRecordMX.prototype._tobj = JkDnsDNSRecordMX;

JkDnsDNSRecordMX.NEW = function() {
	var v = new JkDnsDNSRecordMX;
	return v.CTOR_JkDnsDNSRecordMX();
};

JkDnsDNSRecordMX.prototype.CTOR_JkDnsDNSRecordMX = function() {
	this.priority = 0;
	if(JkDnsDNSRecord.prototype.CTOR_JkDnsDNSRecord.call(this) == null) {
		return null;
	}
	return this;
};

JkDnsDNSRecordMX.prototype.getType = function() {
	return "MX";
};

JkDnsDNSRecordMX.prototype.toString = function() {
	return (JkLangString.safeString((JkDnsDNSRecord.prototype.toString.call(this)))) + (" priority=") + (JkLangString.safeString((JkLangString.forInteger(this.priority)))) + (" ");
};

JkDnsDNSRecordMX.prototype.getPriority = function() {
	return this.priority;
};

JkDnsDNSRecordMX.prototype.setPriority = function(v) {
	this.priority = v;
	return this;
};

JkDnsDNSRecordMX.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkDnsDNSRecordMX"] === true;
};

let JkDnsDNSRecordA = function() {
	JkDnsDNSRecord.call(this);
};

JkDnsDNSRecordA.prototype = (function(o) { function F() {}; F.prototype = o; return(new F()); }) (JkDnsDNSRecord.prototype);
JkDnsDNSRecordA.prototype.constructor = JkDnsDNSRecordA;
JkDnsDNSRecordA.prototype._t = {
	"JkLangStringObject" : true,
	"JkDnsDNSRecordA" : true,
	"JkDnsDNSRecord" : true
};
JkDnsDNSRecordA.prototype._tobj = JkDnsDNSRecordA;

JkDnsDNSRecordA.NEW = function() {
	var v = new JkDnsDNSRecordA;
	return v.CTOR_JkDnsDNSRecordA();
};

JkDnsDNSRecordA.prototype.CTOR_JkDnsDNSRecordA = function() {
	if(JkDnsDNSRecord.prototype.CTOR_JkDnsDNSRecord.call(this) == null) {
		return null;
	}
	return this;
};

JkDnsDNSRecordA.prototype.getType = function() {
	return "A";
};

JkDnsDNSRecordA.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkDnsDNSRecordA"] === true;
};

let JkDnsDNSCacheDNSCacheImpl = function() {
	this.entries = null;
	this.mutex = null;
};

JkDnsDNSCacheDNSCacheImpl.prototype._t = { "JkDnsDNSCacheDNSCacheImpl" : true };
JkDnsDNSCacheDNSCacheImpl.prototype._tobj = JkDnsDNSCacheDNSCacheImpl;

JkDnsDNSCacheDNSCacheImpl.NEW = function() {
	var v = new JkDnsDNSCacheDNSCacheImpl;
	return v.CTOR_JkDnsDNSCacheDNSCacheImpl();
};

JkDnsDNSCacheDNSCacheImpl.prototype.CTOR_JkDnsDNSCacheDNSCacheImpl = function() {
	this.mutex = null;
	this.entries = null;
	this.entries = JkLangDynamicMap.NEW();
	this.mutex = JkThreadMutex.create();
	return this;
};

JkDnsDNSCacheDNSCacheImpl.prototype.add = function(hostname, ip) {
	if(this.mutex != null) {
		this.mutex.lockMutex();
	}
	this.entries.setObject(hostname, (JkDnsDNSCacheDNSCacheEntry.create(ip)));
	if(this.mutex != null) {
		this.mutex.unlockMutex();
	}
};

JkDnsDNSCacheDNSCacheImpl.prototype.getCachedEntry = function(hostname) {
	var v = null;
	if(this.mutex != null) {
		this.mutex.lockMutex();
	}
	v = (function(o) {
		if(JkDnsDNSCacheDNSCacheEntry.IS_INSTANCE && JkDnsDNSCacheDNSCacheEntry.IS_INSTANCE(o)) {
			return o;
		}
		return null;
	}.bind(this))((this.entries.get(hostname)));
	if(this.mutex != null) {
		this.mutex.unlockMutex();
	}
	if(v != null) {
		if(JkTimeSystemClock.asSeconds() - v.getTimestamp() > 60 * 60) {
			if(this.mutex != null) {
				this.mutex.lockMutex();
			}
			this.entries.remove(hostname);
			if(this.mutex != null) {
				this.mutex.unlockMutex();
			}
			v = null;
		}
	}
	if(v != null) {
		return v.getIp();
	}
	return null;
};

JkDnsDNSCacheDNSCacheImpl.prototype.resolve = function(hostname) {
	var v = this.getCachedEntry(hostname);
	if(v != null) {
		return v;
	}
	var dr = JkDnsDNSResolver.create();
	if(!(dr != null)) {
		return null;
	}
	v = dr.getIPAddress(hostname, null);
	if(v != null) {
		this.add(hostname, v);
	}
	return v;
};

JkDnsDNSCacheDNSCacheImpl.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkDnsDNSCacheDNSCacheImpl"] === true;
};

let JkDnsDNSCacheDNSCacheEntry = function() {
	this.ip = null;
	this.timestamp = 0;
};

JkDnsDNSCacheDNSCacheEntry.prototype._t = { "JkDnsDNSCacheDNSCacheEntry" : true };
JkDnsDNSCacheDNSCacheEntry.prototype._tobj = JkDnsDNSCacheDNSCacheEntry;

JkDnsDNSCacheDNSCacheEntry.NEW = function() {
	var v = new JkDnsDNSCacheDNSCacheEntry;
	return v.CTOR_JkDnsDNSCacheDNSCacheEntry();
};

JkDnsDNSCacheDNSCacheEntry.prototype.CTOR_JkDnsDNSCacheDNSCacheEntry = function() {
	this.timestamp = 0;
	this.ip = null;
	return this;
};

JkDnsDNSCacheDNSCacheEntry.create = function(ip) {
	var v = JkDnsDNSCacheDNSCacheEntry.NEW();
	v.setIp(ip);
	v.setTimestamp((JkTimeSystemClock.asSeconds()));
	return v;
};

JkDnsDNSCacheDNSCacheEntry.prototype.getIp = function() {
	return this.ip;
};

JkDnsDNSCacheDNSCacheEntry.prototype.setIp = function(v) {
	this.ip = v;
	return this;
};

JkDnsDNSCacheDNSCacheEntry.prototype.getTimestamp = function() {
	return this.timestamp;
};

JkDnsDNSCacheDNSCacheEntry.prototype.setTimestamp = function(v) {
	this.timestamp = v;
	return this;
};

JkDnsDNSCacheDNSCacheEntry.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkDnsDNSCacheDNSCacheEntry"] === true;
};

let JkDnsDNSCache = function() {};

JkDnsDNSCache.prototype._t = { "JkDnsDNSCache" : true };
JkDnsDNSCache.prototype._tobj = JkDnsDNSCache;

JkDnsDNSCache.NEW = function() {
	var v = new JkDnsDNSCache;
	return v.CTOR_JkDnsDNSCache();
};

JkDnsDNSCache.prototype.CTOR_JkDnsDNSCache = function() {
	return this;
};

JkDnsDNSCache.resolve = function(hostname) {
	if(JkDnsDNSCache.cc == null) {
		JkDnsDNSCache.cc = JkDnsDNSCacheDNSCacheImpl.NEW();
	}
	return JkDnsDNSCache.cc.resolve(hostname);
};

JkDnsDNSCache.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkDnsDNSCache"] === true;
};

JkDnsDNSCache.cc = null;

let JkDnsDNSRecordCNAME = function() {
	JkDnsDNSRecord.call(this);
};

JkDnsDNSRecordCNAME.prototype = (function(o) { function F() {}; F.prototype = o; return(new F()); }) (JkDnsDNSRecord.prototype);
JkDnsDNSRecordCNAME.prototype.constructor = JkDnsDNSRecordCNAME;
JkDnsDNSRecordCNAME.prototype._t = {
	"JkDnsDNSRecordCNAME" : true,
	"JkLangStringObject" : true,
	"JkDnsDNSRecord" : true
};
JkDnsDNSRecordCNAME.prototype._tobj = JkDnsDNSRecordCNAME;

JkDnsDNSRecordCNAME.NEW = function() {
	var v = new JkDnsDNSRecordCNAME;
	return v.CTOR_JkDnsDNSRecordCNAME();
};

JkDnsDNSRecordCNAME.prototype.CTOR_JkDnsDNSRecordCNAME = function() {
	if(JkDnsDNSRecord.prototype.CTOR_JkDnsDNSRecord.call(this) == null) {
		return null;
	}
	return this;
};

JkDnsDNSRecordCNAME.prototype.getType = function() {
	return "CNAME";
};

JkDnsDNSRecordCNAME.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkDnsDNSRecordCNAME"] === true;
};

File: .sling/libs/sushi/eqela/jk-net/0.27.0/jk.dns.lua
jk = jk or {}

jk.dns = jk.dns or {}

jk.dns.DNSRecord = {}
jk.dns.DNSRecord.__index = jk.dns.DNSRecord
_vm:set_metatable(jk.dns.DNSRecord, {})

function jk.dns.DNSRecord._create()
	local v = _vm:set_metatable({}, jk.dns.DNSRecord)
	return v
end

function jk.dns.DNSRecord:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.dns.DNSRecord'
	self['_isType.jk.dns.DNSRecord'] = true
	self['_isType.jk.lang.StringObject'] = true
	self.ttl = 0
	self.address = nil
end

function jk.dns.DNSRecord:_construct0()
	jk.dns.DNSRecord._init(self)
	return self
end

function jk.dns.DNSRecord:getType()
	do return "UNKNOWN" end
end

function jk.dns.DNSRecord:toString()
	do return "type=" .. _g.jk.lang.String:safeString(self:getType()) .. " ttl=" .. _g.jk.lang.String:safeString(_g.jk.lang.String:forInteger(self.ttl)) .. " address=`" .. _g.jk.lang.String:safeString(self.address) .. "'" end
end

function jk.dns.DNSRecord:getTtl()
	do return self.ttl end
end

function jk.dns.DNSRecord:setTtl(v)
	self.ttl = v
	do return self end
end

function jk.dns.DNSRecord:getAddress()
	do return self.address end
end

function jk.dns.DNSRecord:setAddress(v)
	self.address = v
	do return self end
end

jk.dns.DNSRecordA = _g.jk.dns.DNSRecord._create()
jk.dns.DNSRecordA.__index = jk.dns.DNSRecordA
_vm:set_metatable(jk.dns.DNSRecordA, {
	__index = _g.jk.dns.DNSRecord
})

function jk.dns.DNSRecordA._create()
	local v = _vm:set_metatable({}, jk.dns.DNSRecordA)
	return v
end

function jk.dns.DNSRecordA:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.dns.DNSRecordA'
	self['_isType.jk.dns.DNSRecordA'] = true
end

function jk.dns.DNSRecordA:_construct0()
	jk.dns.DNSRecordA._init(self)
	do _g.jk.dns.DNSRecord._construct0(self) end
	return self
end

function jk.dns.DNSRecordA:getType()
	do return "A" end
end

jk.dns.DNSCache = {}
jk.dns.DNSCache.__index = jk.dns.DNSCache
_vm:set_metatable(jk.dns.DNSCache, {})

jk.dns.DNSCache.cc = nil

function jk.dns.DNSCache._create()
	local v = _vm:set_metatable({}, jk.dns.DNSCache)
	return v
end

function jk.dns.DNSCache:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.dns.DNSCache'
	self['_isType.jk.dns.DNSCache'] = true
end

function jk.dns.DNSCache:_construct0()
	jk.dns.DNSCache._init(self)
	return self
end

jk.dns.DNSCache.DNSCacheEntry = {}
jk.dns.DNSCache.DNSCacheEntry.__index = jk.dns.DNSCache.DNSCacheEntry
_vm:set_metatable(jk.dns.DNSCache.DNSCacheEntry, {})

function jk.dns.DNSCache.DNSCacheEntry._create()
	local v = _vm:set_metatable({}, jk.dns.DNSCache.DNSCacheEntry)
	return v
end

function jk.dns.DNSCache.DNSCacheEntry:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.dns.DNSCache.DNSCacheEntry'
	self['_isType.jk.dns.DNSCache.DNSCacheEntry'] = true
	self.ip = nil
	self.timestamp = 0
end

function jk.dns.DNSCache.DNSCacheEntry:_construct0()
	jk.dns.DNSCache.DNSCacheEntry._init(self)
	return self
end

function jk.dns.DNSCache.DNSCacheEntry:create(ip)
	local v = _g.jk.dns.DNSCache.DNSCacheEntry._construct0(_g.jk.dns.DNSCache.DNSCacheEntry._create())
	do v:setIp(ip) end
	do v:setTimestamp(_g.jk.time.SystemClock:asSeconds()) end
	do return v end
end

function jk.dns.DNSCache.DNSCacheEntry:getIp()
	do return self.ip end
end

function jk.dns.DNSCache.DNSCacheEntry:setIp(v)
	self.ip = v
	do return self end
end

function jk.dns.DNSCache.DNSCacheEntry:getTimestamp()
	do return self.timestamp end
end

function jk.dns.DNSCache.DNSCacheEntry:setTimestamp(v)
	self.timestamp = v
	do return self end
end

jk.dns.DNSCache.DNSCacheImpl = {}
jk.dns.DNSCache.DNSCacheImpl.__index = jk.dns.DNSCache.DNSCacheImpl
_vm:set_metatable(jk.dns.DNSCache.DNSCacheImpl, {})

function jk.dns.DNSCache.DNSCacheImpl._create()
	local v = _vm:set_metatable({}, jk.dns.DNSCache.DNSCacheImpl)
	return v
end

function jk.dns.DNSCache.DNSCacheImpl:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.dns.DNSCache.DNSCacheImpl'
	self['_isType.jk.dns.DNSCache.DNSCacheImpl'] = true
	self.entries = nil
	self.mutex = nil
end

function jk.dns.DNSCache.DNSCacheImpl:_construct0()
	jk.dns.DNSCache.DNSCacheImpl._init(self)
	self.entries = _g.jk.lang.DynamicMap._construct0(_g.jk.lang.DynamicMap._create())
	self.mutex = _g.jk.thread.Mutex:create()
	return self
end

function jk.dns.DNSCache.DNSCacheImpl:add(hostname, ip)
	if self.mutex ~= nil then
		do self.mutex:lockMutex() end
	end
	do self.entries:setObject(hostname, _g.jk.dns.DNSCache.DNSCacheEntry:create(ip)) end
	if self.mutex ~= nil then
		do self.mutex:unlockMutex() end
	end
end

function jk.dns.DNSCache.DNSCacheImpl:getCachedEntry(hostname)
	local v = nil
	if self.mutex ~= nil then
		do self.mutex:lockMutex() end
	end
	v = _vm:to_table_with_key(self.entries:get(hostname), '_isType.jk.dns.DNSCache.DNSCacheEntry')
	if self.mutex ~= nil then
		do self.mutex:unlockMutex() end
	end
	if v ~= nil then
		if _g.jk.time.SystemClock:asSeconds() - v:getTimestamp() > 60 * 60 then
			if self.mutex ~= nil then
				do self.mutex:lockMutex() end
			end
			do self.entries:remove(hostname) end
			if self.mutex ~= nil then
				do self.mutex:unlockMutex() end
			end
			v = nil
		end
	end
	if v ~= nil then
		do return v:getIp() end
	end
	do return nil end
end

function jk.dns.DNSCache.DNSCacheImpl:resolve(hostname)
	local v = self:getCachedEntry(hostname)
	if v ~= nil then
		do return v end
	end
	do
		local dr = _g.jk.dns.DNSResolver:create()
		if not (dr ~= nil) then
			do return nil end
		end
		v = dr:getIPAddress(hostname, nil)
		if v ~= nil then
			do self:add(hostname, v) end
		end
		do return v end
	end
end

function jk.dns.DNSCache:resolve(hostname)
	if _g.jk.dns.DNSCache.cc == nil then
		_g.jk.dns.DNSCache.cc = _g.jk.dns.DNSCache.DNSCacheImpl._construct0(_g.jk.dns.DNSCache.DNSCacheImpl._create())
	end
	do return _g.jk.dns.DNSCache.cc:resolve(hostname) end
end

jk.dns.DNSRecordCNAME = _g.jk.dns.DNSRecord._create()
jk.dns.DNSRecordCNAME.__index = jk.dns.DNSRecordCNAME
_vm:set_metatable(jk.dns.DNSRecordCNAME, {
	__index = _g.jk.dns.DNSRecord
})

function jk.dns.DNSRecordCNAME._create()
	local v = _vm:set_metatable({}, jk.dns.DNSRecordCNAME)
	return v
end

function jk.dns.DNSRecordCNAME:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.dns.DNSRecordCNAME'
	self['_isType.jk.dns.DNSRecordCNAME'] = true
end

function jk.dns.DNSRecordCNAME:_construct0()
	jk.dns.DNSRecordCNAME._init(self)
	do _g.jk.dns.DNSRecord._construct0(self) end
	return self
end

function jk.dns.DNSRecordCNAME:getType()
	do return "CNAME" end
end

jk.dns.DNSRecordNS = _g.jk.dns.DNSRecord._create()
jk.dns.DNSRecordNS.__index = jk.dns.DNSRecordNS
_vm:set_metatable(jk.dns.DNSRecordNS, {
	__index = _g.jk.dns.DNSRecord
})

function jk.dns.DNSRecordNS._create()
	local v = _vm:set_metatable({}, jk.dns.DNSRecordNS)
	return v
end

function jk.dns.DNSRecordNS:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.dns.DNSRecordNS'
	self['_isType.jk.dns.DNSRecordNS'] = true
end

function jk.dns.DNSRecordNS:_construct0()
	jk.dns.DNSRecordNS._init(self)
	do _g.jk.dns.DNSRecord._construct0(self) end
	return self
end

function jk.dns.DNSRecordNS:getType()
	do return "NS" end
end

jk.dns.DNSRecordMX = _g.jk.dns.DNSRecord._create()
jk.dns.DNSRecordMX.__index = jk.dns.DNSRecordMX
_vm:set_metatable(jk.dns.DNSRecordMX, {
	__index = _g.jk.dns.DNSRecord
})

function jk.dns.DNSRecordMX._create()
	local v = _vm:set_metatable({}, jk.dns.DNSRecordMX)
	return v
end

function jk.dns.DNSRecordMX:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.dns.DNSRecordMX'
	self['_isType.jk.dns.DNSRecordMX'] = true
	self.priority = 0
end

function jk.dns.DNSRecordMX:_construct0()
	jk.dns.DNSRecordMX._init(self)
	do _g.jk.dns.DNSRecord._construct0(self) end
	return self
end

function jk.dns.DNSRecordMX:getType()
	do return "MX" end
end

function jk.dns.DNSRecordMX:toString()
	do return _g.jk.lang.String:safeString(_g.jk.dns.DNSRecord.toString(self)) .. " priority=" .. _g.jk.lang.String:safeString(_g.jk.lang.String:forInteger(self.priority)) .. " " end
end

function jk.dns.DNSRecordMX:getPriority()
	do return self.priority end
end

function jk.dns.DNSRecordMX:setPriority(v)
	self.priority = v
	do return self end
end

jk.dns.DNSResolver = {}
jk.dns.DNSResolver.__index = jk.dns.DNSResolver
_vm:set_metatable(jk.dns.DNSResolver, {})

function jk.dns.DNSResolver._create()
	local v = _vm:set_metatable({}, jk.dns.DNSResolver)
	return v
end

function jk.dns.DNSResolver:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.dns.DNSResolver'
	self['_isType.jk.dns.DNSResolver'] = true
end

function jk.dns.DNSResolver:_construct0()
	jk.dns.DNSResolver._init(self)
	return self
end

function jk.dns.DNSResolver:create()
	do return nil end
end

function jk.dns.DNSResolver:instance()
	do return _g.jk.dns.DNSResolver:create() end
end

function jk.dns.DNSResolver:getNSRecords(host, type, ctx)
end

function jk.dns.DNSResolver:getIPAddress(hostname, ctx)
end

function jk.dns.DNSResolver:getIPAddresses(hostname, ctx)
end
